logstash-core/lib/logstash/runner.rb

```diff
@@ -601,4 +604,16 @@ def setting(key)
     @settings.get_value(key)
   end
 
+  def configure_io_receive_buffer_allocation_to_heap
+    # if user hasn't explicitly set Netty's interested properties
+    if java.lang.System.getProperty("io.netty.allocator.numDirectArenas") == nil &&
+        java.lang.System.getProperty("io.netty.noPreferDirect")  == nil
+      # set Netty's Java properties to disable direct memory allocation
```

Comment:
We should soften the language here. `noPreferDirect`, as the name says, it's just setting the preference. it doesn't prevent direct memory allocation.